src/Cake.SqlServer/Backup/BackupDatabaseImpl.cs

```diff
@@ -28,21 +31,31 @@ internal static void Execute(ICakeContext context, string connectionString, stri
 
                 context.Log.Information(sql);
 
-                var command = SqlServerAliasesImpl.CreateSqlCommand(sql, connection);
-                command.Parameters.AddWithValue("@backupFile", backupFile);
-                command.ExecuteNonQuery();
+                using (var command = SqlServerAliasesImpl.CreateSqlCommand(sql, connection))
+                {
+                    command.Parameters.AddWithValue("@backupFile", backupFile);
+                    command.ExecuteNonQuery();
+                }
             }
         }
 
-        private static string GetBackupFileName(string databaseName, string path)
+        private static string GetBackupFileName(string databaseName, string? path)
         {
-            var fileName = databaseName.Replace("[", string.Empty).Replace("]", string.Empty) + ".bak";
+#if NET5_0
+            var fileName = databaseName
+                .Replace("[", string.Empty, StringComparison.OrdinalIgnoreCase)
+                .Replace("]", string.Empty, StringComparison.OrdinalIgnoreCase) + ".bak";
+#else
+            var fileName = databaseName
+                .Replace("[", string.Empty)
```

Comment:
Do we need to split into .net5 and other? should we rather always do `OrdinalIgnoreCase`?

Reply:
I would add this everywhere - unfortunately that api appeared in .net core 2.0 - it is not existent in full .net framework nor net standard 2.0. Please see
https://docs.microsoft.com/pl-pl/dotnet/api/system.string.replace?view=netstandard-2.0
https://docs.microsoft.com/pl-pl/dotnet/api/system.string.replace?view=netframework-4.8
https://docs.microsoft.com/pl-pl/dotnet/api/system.string.replace?view=netcore-2.0

Reply:
Ah, cool. Did not realise this is not available on .net framework